lib/Provider/Command/MessageSend.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+namespace OCA\Mail\Provider\Command;
+
+use OCA\Mail\Db\LocalMessage;
+use OCA\Mail\Service\AccountService;
+use OCA\Mail\Service\Attachment\AttachmentService;
+use OCA\Mail\Service\OutboxService;
+use OCP\AppFramework\Utility\ITimeFactory;
+use OCP\IConfig;
+use OCP\Mail\Provider\Exception\SendException;
+use OCP\Mail\Provider\IAddress;
+use OCP\Mail\Provider\IMessage;
+
+class MessageSend {
+
+	public function __construct(
+		protected IConfig $config,
+		protected ITimeFactory $time,
+		protected AccountService $accountService,
+		protected OutboxService $outboxService,
+		protected AttachmentService $attachmentService
+	) {
+	}
+
+	/**
+	 * Performs send operation
+	 *
+	 * @since 4.0.0
+	 *
+	 * @param string $userId system user id
+	 * @param string $serviceId mail account id
+	 * @param IMessage $message mail message object with all required parameters to send a message
+	 * @param array $options array of options reserved for future use
+	 *
+	 * @return LocalMessage
+	 * 
+	 * @throws SendException on failure, check message for reason
+	 * 
+	 */
+	public function perform(string $userId, string $serviceId, IMessage $message, array $options = []): LocalMessage {
+		// find user mail account details
+		$account = $this->accountService->find($userId, (int)$serviceId);
```

Comment:
Why exactly do you want me to re-throw database errors? The SendExceptions are for logical sending error (like missing content in an attachment or missing to address) they are not a catch all for every error we can have in the system

Reply:
I don't want you to re-throw the database error, the interface/ocp api does ;)

https://github.com/nextcloud/server/blob/7362ede426becc20e0f2285b72b8e5fec3edfb70/lib/public/Mail/Provider/IMessageSend.php#L21-L30

The interface says, there's a method sendMessage and it might throw a SendException. That's the contract other apps/server develop against. How should an implementation in server know, without checking the code in the mail app, that a ClientException is thrown? 

As last resort the implementation can always catch Throwable, but then the interface should also document it.

Reply:
Done